app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/models/db/DataOpenHelper.kt

```diff
@@ -0,0 +1,689 @@
+                    cal.add(Calendar.MINUTE, 1)
+                } while (cursor.moveToNext())
+            }
+            db.setTransactionSuccessful()
+        } catch (e: SQLException) {
+            Timber.e(e, "Migrating database to version 2")
+        } finally {
+            try {
+                db.endTransaction()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+
+            try {
+                cursor?.close()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+        }
+    }
+
+    private fun updateLatestEpisodeFields(db: SQLiteDatabase) {
+        try {
+            with(db) {
+                beginTransaction()
+                execSQL("ALTER TABLE podcast ADD latest_episode_date INTEGER", arrayOf())
+                execSQL("UPDATE podcast SET latest_episode_uuid = NULL", arrayOf())
+                setTransactionSuccessful()
+            }
+        } catch (e: SQLException) {
+            Timber.e(e, "Migrating database to version 2")
+        } finally {
+            try {
+                db.endTransaction()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+        }
+    }
+
+    private fun clearPodcastTintColours(db: SQLiteDatabase) {
+        db.execSQL("UPDATE podcast SET primary_color = NULL, secondary_color = NULL")
+    }
+
+    @Suppress("SameParameterValue")
+    private fun maxColumn(db: SQLiteDatabase, table: String, column: String): Int {
+        val result = firstRowArray("SELECT MAX($column) FROM $table", arrayOf(), db)
+        return result?.filterNotNull()?.first()?.toInt() ?: 0
+    }
+
+    @Suppress("SameParameterValue")
+    private fun count(db: SQLiteDatabase, name: String, where: String?): Int {
+        val whereSql = if (!where.isNullOrEmpty()) " WHERE $where" else ""
+        val result = firstRowArray("SELECT count(*) FROM $name$whereSql", arrayOf(), db)
+        return result?.first()?.toInt() ?: 0
+    }
+
+    private fun firstRowArray(query: String, params: Array<String?>, db: SQLiteDatabase): Array<String?>? {
+        var cursor: Cursor? = null
+        try {
+            cursor = db.rawQuery(query, params)
+            if (cursor.moveToNext()) {
+                val result = arrayOfNulls<String>(cursor.columnCount)
+                for (i in 0 until cursor.columnCount) {
+                    result[i] = cursor.getString(i)
+                }
+                return result
+            } else {
+                return null
+            }
+        } finally {
+            close(cursor)
+        }
+    }
+
+    private fun close(cursor: Cursor?) {
+        if (cursor == null) return
+
+        try {
+            cursor.close()
+        } catch (exception: java.lang.Exception) {
+            Timber.e(exception)
+        }
+    }
```

Comment:
This will probably become obsolete if you migrate to `Closable.use()` on `Cursor`.